packages/notifi-react-card/lib/hooks/useUnreadState.ts

```diff
@@ -30,27 +30,20 @@ export const useUnreadState = () => {
   }, [frontendClient.userState?.status]);
 
   useEffect(() => {
-    if (!walletPublicKey || !isClientAuthenticated) return;
+    if (!walletPublicKey || !isClientAuthenticated || isStateChangedSubscribed) return;
 
-    frontendClient
-      .getUnreadNotificationHistoryCount()
-      .then((res) => {
-        const unreadNotificationCount = res.count;
-        setUnreadNotificationCount(unreadNotificationCount);
-      })
-      .catch((_e) => {
-        /* Intentionally empty (Concurrent can only possibly happens here instead of inside interval) */
-      });
-    const interval = setInterval(() => {
-      if (!walletPublicKey || !isClientAuthenticated) return;
+    console.log("Called: useUnreadState.subscribeNotificationHistoryStateChanged for user:");
+    setIsStateChangedSubscribed(true);
+
+    frontendClient.subscribeNotificationHistoryStateChanged(() => {
+
+      console.log("CallBack: subscribeNotificationHistoryStateChanged");
 
       frontendClient.getUnreadNotificationHistoryCount().then((res) => {
         const unreadNotificationCount = res.count;
         setUnreadNotificationCount(unreadNotificationCount);
       });
-    }, Math.floor(Math.random() * 5000) + 5000); // a random interval between 5 and 10 seconds to avoid spamming the server
-
-    return () => clearInterval(interval);
+    });
```

Comment:
Not sure if we want to close the websocket after the component is unmounted? Otherwise, will it cause multiple wss connections when the component is rendered multiple times?